### PR TITLE
Blocked mlp single layer in bf16

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**

### CODE REVIEW

**Comment by rengolin** [comment]
The test failure is not related to the code here and is super weird. @hfp any ideas?

**Comment by rengolin** [comment]
Otherwise, patch looks good. I'll let @chelini look at the IR shape to spot anything I missed.

File: test/BF16/mlp-single-layer-blocked-bf16.mlir [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//RUN:standalone-opt %s -linalg-ext-to-loops -convert-linalg-to-loops -convert-tpp-to-xsmm -convert-xsmm-to-func -convert-vector-to-scf -convert-scf-to-cf -convert-vector-to-llvm -convert-func-to-llvm -convert-memref-to-llvm -canonicalize -sparse-compiler |\
```
Comment (chelini) [inline test/BF16/mlp-single-layer-blocked-bf16.mlir:1]: please add a space.

File: test/BF16/mlp-single-layer-blocked-bf16.mlir [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// RUN: mlir-cpu-runner \
```
Comment (chelini) [inline test/BF16/mlp-single-layer-blocked-bf16.mlir:2]: Can you please check the test result?

File: include/Standalone/Dialect/Tpp/TppOps.td [hunk 3]
```diff
@@ -25,6 +25,7 @@ class StaticMemRefRankOf<list<Type> allowedTypes, list<int> ranks> :
 def TppMemRef : StaticMemRefRankOf<[AnyFloat], [1, 2]>;
 def TppPackedMemrefInput : StaticMemRefRankOf<[AnyFloat], [1, 2, 3]>;
 def TppBRGEMMemrefInput : StaticMemRefRankOf<[AnyFloat], [3]>;
+def TppBRGEMMPackedMemrefInput : StaticMemRefRankOf<[AnyFloat], [3,4]>;
```
Comment (chelini) [inline include/Standalone/Dialect/Tpp/TppOps.td:28]: Can you please add verifier checks for this change? Similar with what you did for gemm.

Comment (chelini) [inline include/Standalone/Dialect/Tpp/TppOps.td:28]: Also IREE direction for handling bf16 is maybe a good idea here. We are opening up this op too much. But let's discuss later and ignore this comment for now.

**Review by chelini — CHANGES_REQUESTED** [review]

**Review by chelini — APPROVED** [review]

**Pull request merged**